transport/src/main/java/io/netty5/channel/AbstractChannel.java

```diff
@@ -1299,22 +1289,31 @@ protected final ChannelOutboundBuffer outboundBuffer() {
     protected abstract void doRead(boolean wasReadPendingAlready) throws Exception;
 
     /**
-     * Flush the content of the given buffer to the remote peer.
+     * Called in a loop when writes should be performed until this methods returns {@code false} or there are
+     * no more messages to write.
+     *
+     * @param writeSink                             the {@link WriteSink} that must be completed with the write
+     *                                              progress.
+     *                                              {@link WriteSink#complete(long, long, int, boolean)} or
+     *                                              {@link WriteSink#complete(long, Throwable, boolean)} must be called
+     *                                              exactly once before this method returns non-exceptional.
+     * @throws Exception                            if and error happened during writing.
      */
-    protected abstract void doWrite(ChannelOutboundBuffer in,  WriteHandleFactory.WriteHandle writeHandle)
-            throws Exception;
+    protected abstract void doWriteNow(WriteSink writeSink) throws Exception;
 
     /**
      * Connect to remote peer.
      *
      * @param remoteAddress     the address of the remote peer.
      * @param localAddress      the local address of this channel.
+     * @param initialData       the initial data that is written during connect
+     *                          (if {@link ChannelOption#TCP_FASTOPEN_CONNECT} is supported)
      * @return                  {@code true} if the connect was completed, {@code false} if {@link #finishConnect()}
      *                          will be called later again to try finishing the connect.
      * @throws Exception        thrown on error.
      */
     protected abstract boolean doConnect(
-            SocketAddress remoteAddress, SocketAddress localAddress) throws Exception;
+            SocketAddress remoteAddress, SocketAddress localAddress, Buffer initialData) throws Exception;
```

Comment:
@chrisvest we could also use `Object initalData` here... WDYT ?

Reply:
not sure this is true in general.. Sure it is for sockets but for a "custom" channel implementation it might be not true.

Reply:
Fair point, we can make it `Object`